### PR TITLE
Check for an internal redirect URL

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Check for an internal redirect URL

### Testing Instructions

Review this please

### Expected result

You are not redirected to a external URL even if it is passed via the option as base64 decoded string.

### Actual result

You are redirected to a external URL if there is a base64 decoded string passed.

### Documentation Changes Required

None.

### CODE REVIEW

File: src/components/com_weblinks/models/form.php [hunk 1]
```diff
@@ -35,7 +35,16 @@ class WeblinksModelForm extends WeblinksModelWeblink
 	 */
 	public function getReturnPage()
 	{
-		return base64_encode($this->getState('return_page'));
+		$returnPageUrl = base64_encode($this->getState('return_page'));
+
+		if (!empty($returnPageUrl) && JUri::isInternal($redirectUrl))
```
Comment (astridx) [inline src/components/com_weblinks/models/form.php]: Is the name $redirectUrl right or should this be $returnPageUrl?

File: src/components/com_weblinks/models/form.php [hunk 2]
```diff
@@ -35,7 +35,16 @@ class WeblinksModelForm extends WeblinksModelWeblink
...
+		{
+			return $returnPageUrl;
+		}
+
+		// If the page is not internal or empty return to the home page
+		return JUri::base();
+
```
Comment (astridx) [inline src/components/com_weblinks/models/form.php]: Is the empty line according our conding standards?

File: src/components/com_weblinks/models/form.php (hunk 1 above)
Comment (zero-24) [inline src/components/com_weblinks/models/form.php]: Oops. Thanks i'm going to fix this when i'm back at home.

**Comment by zero-24** [comment]
fixed with the last commit thanks!

**Comment by zero-24** [comment]
@yvesh can you please review & merge?

**Pull request closed without merging**